Makefile

```diff
@@ -23,7 +24,7 @@ build_cpu_image:
 	docker build --build-arg IMAGE=optimum-benchmark:latest-cpu --build-arg USER_ID=$(USER_ID) --build-arg GROUP_ID=$(GROUP_ID) -t optimum-benchmark:latest-cpu docker/unroot
 
 build_cuda_image:
-	docker build -t optimum-benchmark:latest-cuda docker/cuda
+	docker build --build-arg TORCH_VERSION=$(TORCH_VERSION) -t optimum-benchmark:latest-cuda docker/cuda
```

Comment:
I think you can suppress the torch version by first installing `autoawq` (it will install whatever version it wants) then install latest torch version, then compile awq kernels.

Reply:
“AutoAWQ_kernels also requires torch==2.3.1. (https://github.com/casper-hansen/AutoAWQ_kernels/blob/main/setup.py) so this cannot work directly.

What I could do is perform a full installation with`torch==2.3.1` for AutoAWQ and AutoAWQ_kernels, then install the latest version of torch, and proceed with the rest of the package installation using the latest version of PyTorch.

However, I see this as an issue for two reasons:
1) The installation process would be a bit more convoluted. Indeed, the setup.py installation cannot handle these requirements directly (installing one version for a given package and then another version afterward). I would need to do something like `pip install . -e [autoawq]`, then `pip install --no-cache-dir torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121`, and finally `pip install -e .[testing,timm,diffusers,peft,auto-gptq,bitsandbytes,deepspeed]`. This process could be abstracted away in the Makefile, but it’s still not the prettiest solution.
2) I think this could cause problems later, as AutoAWQ might be incompatible with newer versions of PyTorch.

What do you think?

Reply:
AutoAWQ kernels will definitely be incompatible, `torch` doesn't guarantee anything about extensions compatibility.
Since we're gonna build from source either ways, it's just one additional command to remove the pin from `setup.py` once it's cloned:
```bash
git clone https://github.com/casper-hansen/AutoAWQ.git
sed -i 's/torch==[0-9\.]*/torch/' AutoAWQ_kernels/setup.py
```

Reply:
Great idea! I modified it to implement your solution. I used a regex instead of sed, but the results are similar. Also, I couldn’t directly add the AutoAWQ installation to the setup.py (as it doesn’t support this use case), so I added a Python script to install the dependency